planner/datasource.go

```diff
@@ -43,7 +43,7 @@ func (p *Planner) getCollectionScanPlan(parsed *mapper.Select) (planSource, erro
 	}
 
 	scan := p.Scan(parsed)
-	err = scan.initCollection(colDesc)
+	err = scan.initCollection(colDesc) // @todo: Should `initCollection` be merged into `p.Scan(...)`
```

Comment:
todo: I have an very large preference for this todo to be done.  I guess it is technically out of scope, but `initCollection` is amusingly horrible (presumably from a time where unsure whether it would be shared/re-inited) and it would be quick todo.

Failing that, please make sure that either the todo is removed, or is bound to a ticket, before you merge this PR